internal/metrics/otel_label.go

```diff
@@ -0,0 +1,48 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package metrics
+
+import "go.opentelemetry.io/otel/attribute"
+
+// A Label provides a named dimension for a Metric.
+type Label struct {
```

Comment:
is there any reason not directly use types from OTel?

Reply:
We use a wrapper around metrics to hide complexity of otel lib. Take otel gauge as an example, creating a otel gauge for now is complex, we need to implement observable callback. And also other advantages besides easy-to-use, like we can do some optimization but in high level use, we can hide the details.

As for label, why creating a wrapper around it, I don't think we need to import two packages: we need both metrics lib of our own and otel attributes lib, it is weird to be use when creating metrics along with some labels.